numpy/array_api/_data_type_functions.py

```diff
@@ -98,7 +98,7 @@ def iinfo(type: Union[Dtype, Array], /) -> iinfo_object:
     return iinfo_object(ii.bits, ii.max, ii.min)
 
 
-def result_type(*arrays_and_dtypes: Sequence[Union[Array, Dtype]]) -> Dtype:
+def result_type(*arrays_and_dtypes: Union[Array, Dtype]) -> Dtype:
```

Comment:
With variadic parameters the annotation applies to every individual parameter, rather than a to-be unpacked iterable as a whole.

Reply:
Should we update this in the spec?

Reply:
I'd say so, yes, as both type of annotations have a rather different meaning.

Reply:
The spec has been updated as of https://github.com/data-apis/array-api/pull/276.